keras_nlp/layers/modeling/lora_dense.py

```diff
@@ -0,0 +1,230 @@
+# Copyright 2023 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import re
+
+from keras_nlp.api_export import keras_nlp_export
+from keras_nlp.backend import config
+from keras_nlp.backend import keras
+from keras_nlp.backend import ops
+
+
+def validate_einsum_equation(equation):
+    left, right, output = re.split(",|->", equation)
+    # For simplicity, we greatly restrict possible einsum equations. The final
+    # axis of the input must be the first axis of our kernel, and must not
+    # appear in our output.
+    valid = left[-1] == right[0] and left[-1] not in output
+    if not valid:
+        raise ValueError(
+            "When passing a `EinsumDense` layer to a `LoraDense` layer, the "
+            "einsum `equation` must always have the form `*x,x*->*`, where "
+            "each `*` can be any sequence. Conceptually, the `equation` should "
+            "always represent a dense matmul on the last axis of the input. "
+            f"Received invalid equation `'{equation}'`."
+        )
+
+
+@keras_nlp_export("keras_nlp.layers.LoraDense")
+class LoraDense(keras.layers.Layer):
```

Comment:
I'm not opinionated, just want to make sure it's considered -- should this ever be LoRADense? IIRC the Keras convention is to exclude this sort of capitalization but I can't remember off the top of my head.

Reply:
Yeah, when I discussed this with Francois we landed on if it is pronounceable and colloquially said, consider it a single segment in our camel casing. So "Roberta" is just that, not RoBERTa. Honestly it would be a bit of a homework assignment to remeber where all the casing goes.

We actually have a line for this in our style guide! https://github.com/keras-team/keras-nlp/blob/master/STYLE_GUIDE.md#naming-of-layers-and-models